forem/forem
# Member Index View Actions: "Banish user"

@Ridhwana responsible for taking this PR to completion. 

## What type of PR is this? (check all applicable)


## Description
This PR adds a "Banish user" modal to the member's action panel in the Member Index View by leveraging the reusable modal. Additionally, this PR adds makes the `Admin::UsersController#set_banishable_user` method available on the index page, conditionally renders the "Banish user" button for only users who have not yet been banished, and adds e2e tests behind the `member_index_view` feature flag that test that the modal shows as expected.

## Related Tickets & Documents
- Closes #17486 

## QA Instructions, Screenshots, Recordings
To QA this PR, please first ensure that you have the `member_index_view` feature flag enabled, and then navigate to `/admin/member_manager/users`.

Next, test that you are able to open the "Banish user" modal, located within the vertical menu in the righthand corner of each member, by clicking the "Banish user" button within the menu. This button should only show when the user has not yet been banished, so you should also test that the button conditionally renders as expected.

After, test that you are able to banish a user as expected and that upon banishing the user, you are redirected to the view from which you banished the user, (either `/admin/member_manager/users` or `/admin/member_manager/users/user_id`).

Note to mimic a user that will be banishable, you can do the following: 
```
u = User.find(:id)
u.created_at = 2.years.ago
u.save!
```


This should work for screens of all sizes.

#### After (Non-Banished User):
![Screen Shot 2022-06-24 at 1 10 20 PM](https://user-images.githubusercontent.com/32834804/175649843-53a5e3cf-469b-4b0c-b54a-fc69f640ea11.png)

#### After (Banished User):
![Screen Shot 2022-06-24 at 1 10 38 PM](https://user-images.githubusercontent.com/32834804/175649873-54ffa67d-6bfc-4f96-8f99-2fd36f90c0ef.png)

### UI accessibility concerns?
There shouldn't be! 🤞 

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
None.

## [optional] What gif best describes this PR or how it makes you feel?

![Two people saying: "Ya banished!" and snapping their fingers](https://media.giphy.com/media/KxcHCxYZkKfnG6EKPT/giphy.gif)
